tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/RoSXMLHelper.cs

```diff
@@ -0,0 +1,239 @@
+
+
+            string[] fieldNames = { "LICNO","FLEET" };
+            string[] fieldNamesProduct = { "REPEAT" };
+            Z_ADDS_ROSIM_ORDERItem[] items = roSXmlField.PROD;
+            JsonInputRoSWebhookHelper.Unitsofsale[] unitofsales = roSJsonFields.unitsOfSale;
+            VerifyBlankFields(roSXmlField, fieldNames);
+            VerifyProductFields(items, unitofsales);
+            VerifyBlankProductFields(items, fieldNamesProduct);
+            if (s_attrNotMatched.Count == 0)
+            {
+                Console.WriteLine("NEWLICENCE event XML is correct");
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("NEWLICENCE event XML is incorrect");
+                Console.WriteLine("Not matching attributes are:");
+                foreach (string attribute in s_attrNotMatched)
+                { Console.WriteLine(attribute); }
+                return false;
+            }
+        }
+
+        private static void VerifyBlankFields(Z_ADDS_ROSIM_ORDER roSXmlField, string[] fieldNames)
+        {
+            foreach (string field in fieldNames)
+            {
+                if (!typeof(Z_ADDS_ROSIM_ORDER).GetProperty(field).GetValue(roSXmlField, null).Equals(""))
+                    s_attrNotMatched.Add(typeof(Z_ADDS_ROSIM_ORDER).GetProperty(field).Name);
+            }
+        }
+
+        private static void VerifyProductFields(Z_ADDS_ROSIM_ORDERItem[] roSResult, JsonInputRoSWebhookHelper.Unitsofsale[] unitofsales)
+        {
+            int i = 0;
+            foreach (Z_ADDS_ROSIM_ORDERItem prodxml in roSResult)
+            {
+                if (!prodxml.ID.Equals(unitofsales[i].unitName))
+                        s_attrNotMatched.Add(nameof(prodxml.ID));
+                if (!prodxml.ENDDA.Equals(unitofsales[i].endDate))
+                        s_attrNotMatched.Add(nameof(prodxml.ENDDA));
+                if (!prodxml.DURATION.Equals(unitofsales[i].duration))
+                        s_attrNotMatched.Add(nameof(prodxml.DURATION));
+                if (!prodxml.RENEW.Equals(unitofsales[i].renew))
+                        s_attrNotMatched.Add(nameof(prodxml.RENEW));
+                if (!prodxml.REPEAT.Equals(unitofsales[i].repeat))
+                        s_attrNotMatched.Add(nameof(prodxml.REPEAT));
+                i++; 
+            }
+                
+            
+        }
+
+        public static async Task<bool> VerifyPresenseOfMandatoryXMLAtrributes(Z_ADDS_ROSIM_ORDER order)
+        {
+            List<string> ActionAttributesSeq = new();
+            ActionAttributesSeq = Config.TestConfig.RosLicenceUpdateXMLList.ToList<string>();
+            List<string> currentActionAttributes = new();
+            currentActionAttributes.Clear();
+            Type arrayType = order.GetType();
+            System.Reflection.PropertyInfo[] properties = arrayType.GetProperties();
+            foreach (System.Reflection.PropertyInfo property in properties)
```

Comment:
```suggestion
currentActionAttributes.AddRange(properties.Select(property => property.Name));
```

Reply:
Can you try this option instead of writing foreach loop.

Reply:
Done